### ISSUE TITLE
SPFx ListView Command Set extension in Microsoft Teams - Workaround

### ISSUE THREAD

**Issue by necik11** [pnp/pnpjs#3092]
### What version of PnPjs library you are using

3.x

### Minor Version Number

3.20.1

### Target environment

All

### Additional environment details

I created a web part that I want to use in Teams or on the web.

### Question/Request

Is there a workaround to run Comand set - extension as a webpart without having to create my own grid webpart?
I need to run Command over an item / document and the only thing I could think of was to create my own web part and in it there would be a custom datagrid view + click event = full of work.
Second one idea is create input, where the user would enter the id of the item and then the item would be loaded.

Also, it's quite annoying when I have to enter input instead of clicking one. [Issue](https://github.com/SharePoint/sp-dev-docs/issues/9275).

It is difficult to explain to the customer that he has to click open in sharepoint from the team and work there... If there is something better than what I described, I will be happy for any tip.

**Comment by juliemturner** [comment]
This is not really a PnPjs question. There are a lot of forums out there that would be better place to post this question. I will say this though, command sets load in Teams, so if you have a command set deployed to a list/library and then pin that list/library as a tab in teams the command set will work.

**Comment by necik11** [comment]
I use pnpjs + spfx, and I didn't know that Comand set - extension also works in teams, so I'm trying to find a solution with the help of PnPJS.

Anyway, what I tried, after creating the Command set, I set 100 and 101 and neither of them works in teams if I add a tab as a sheet or a document library.
I was looking in debug mode in Chrome and got on all CA = blocked by CSP. 
`https://{tenant}.sharepoint.com/sites/apps/ClientSideAssets/{id}/hello-command-set_d35e7fc51ab419d673a5.js`
strict-origin-when-cross-origin

However when I add the library/sheet as a SharePoint .aspx page. The CA starts to appear, but unfortunately, the rest API does not work, because: this._context.pageContext.list.id;
show library: SitePages sheet.

And Again, I don't know if I should ask the SPFx forum about verifying "blocked by CSP." or webpart in .aspx = workaround `this._context.pageContext.list.id` is wrongly loaded into PnP...

My assumption is that SPFx should guarantee the operation of the Command Set of the library/sheet without a workaround, and therefore I should direct the question there.

I apologize if I have delayed you in any way. SPFx is closely related to PnPJS and sometimes I am confused about where to ask.
